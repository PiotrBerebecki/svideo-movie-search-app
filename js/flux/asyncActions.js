import axios from 'axios';
import { addApiData } from './actions';

export default function getApiDetails(imdbID) {
  return dispatch => {
    axios
      .get(`http://localhost:3000/${imdbID}`)
      .then(response => {
        dispatch(addApiData(response.data));
      })
      .catch(err => {
        console.error('axios error', err);
      });
  };
}
