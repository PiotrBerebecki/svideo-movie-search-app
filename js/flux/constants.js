export const SET_SEARCH_TERM = 'SET_SEARCH_TERM';
export const SET_SEARCH_TERM1 = 'SET_SEARCH_TERM';
// export const SET_SEARCH_TERM4 = 'SET_SEARCH_TERM';
