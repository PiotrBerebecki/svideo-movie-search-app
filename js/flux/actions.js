import { SET_SEARCH_TERM } from './constants';

export function setSearchTerm(searchTerm) {
  return {
    type: SET_SEARCH_TERM,
    payload: searchTerm,
  };
}

export function setSearchTermming(searchTerm) {
  return {
    type: 'SET_SEARCH_TERMING',
    payload: searchTerm,
  };
}
