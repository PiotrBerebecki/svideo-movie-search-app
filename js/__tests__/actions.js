import { setSearchTerm, addApiData, getApiDetails } from './../flux/actions';
import moxios from 'moxios';

describe('actions', () => {
  it('setSearchTerm', () => {
    const searchTerm = 'black';
    expect(setSearchTerm(searchTerm)).toEqual({
      type: 'SET_SEARCH_TERM',
      payload: searchTerm,
    });
  });

  it('setSearchTerm - using SnapShot', () => {
    expect(setSearchTerm('black')).toMatchSnapshot();
  });

  const atlanta = {
    title: 'Atlanta',
    year: '2008â€“2013',
    description: 'Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; "Earnest \'Earn\' Marks," an ambitious college drop-out and his estranged cousin, who suddenly becomes a star.',
    poster: 'a.jpg',
    imdbID: 'tt4288182',
    trailer: 'MpEdJ-mmTlY',
    ratings: '4.0',
  };

  it('addApiData - using SnapShot', () => {
    expect(addApiData(atlanta)).toMatchSnapshot();
  });

  it('getApiDetails', done => {
    const dispatchMock = jest.fn();

    moxios.withMock(() => {
      const thunk = getApiDetails(atlanta.imdbID);
      thunk(dispatchMock);
      moxios.wait(() => {
        const request = moxios.requests.mostRecent();
        request
          .respondWith({
            status: 200,
            response: atlanta,
          })
          .then(() => {
            expect(request.url).toEqual(
              `http://localhost:3000/${atlanta.imdbID}`
            );
            expect(dispatchMock).toBeCalledWith(addApiData(atlanta));
            done();
          });
      });
    });
  });
});
