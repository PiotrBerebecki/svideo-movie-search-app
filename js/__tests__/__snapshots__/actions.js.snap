// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions addApiData - using SnapShot 1`] = `
Object {
  "payload": Object {
    "description": "Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; \\"Earnest 'Earn' Marks,\\" an ambitious college drop-out and his estranged cousin, who suddenly becomes a star.",
    "imdbID": "tt4288182",
    "poster": "a.jpg",
    "ratings": "4.0",
    "title": "Atlanta",
    "trailer": "MpEdJ-mmTlY",
    "year": "2008â€“2013",
  },
  "type": "ADD_API_DATA",
}
`;

exports[`actions setSearchTerm - using SnapShot 1`] = `
Object {
  "payload": "black",
  "type": "SET_SEARCH_TERM",
}
`;
